spec/models/principal_spec.rb

```diff
@@ -202,4 +202,325 @@
       expect(Firm.where(id: firm.id)).to be_empty
     end
   end
+
+  describe '#next_onboarding_action' do
+    context 'when principal has no parent firm' do
+      before :each do
+        principal.firm.destroy
+        principal.reload
+      end
+
+      it 'returns :complete_a_firm' do
+        expect(principal.next_onboarding_action).to eql(:complete_a_firm)
+      end
+    end
+
+    context 'when principal only has a parent firm' do
+      before :each do
+        expect(principal.firm).not_to be_nil
+        expect(Firm.where(fca_number: principal.fca_number).count).to eql(1)
+      end
+
+      context 'and the parent firm has no email address' do
```

Comment:
I agree with this. It's something that we struggled with when writing these tests.